### ISSUE TITLE
Overplot_lines and overplot_points do not deal with reversed corner plots

### ISSUE THREAD

**Issue by NeutralKaon** [dfm/corner.py#155]
Hi there, 

If you create a reversed corner plot using `reverse=true`, the helpful and new `overplot_` functions do not account for this and merrily put their output in the non-existent axes in the transpose of the graph. 

A simple MWE of the problem would be: 

```python3
#Taken from the docs 
import corner
import numpy as np

ndim, nsamples = 4, 50000
np.random.seed(1234)
data1 = np.random.randn(ndim * 4 * nsamples // 5).reshape([4 * nsamples // 5, ndim])
mean = 4*np.random.rand(ndim)
data2 = (mean[None, :] + np.random.randn(ndim * nsamples // 5).reshape([nsamples // 5, ndim]))
samples = np.vstack([data1, data2])

value1 = mean
# This is the empirical mean of the sample:
value2 = np.mean(samples, axis=0)

###################################################
# Make the base corner plot -- but with reverse=True!
###################################################
figure = corner.corner(samples,reverse=True)

corner.overplot_lines(figure, value1, color="C1")
corner.overplot_points(figure, value1[None], marker="s", color="C1")
corner.overplot_lines(figure, value2, color="C2")
corner.overplot_points(figure, value2[None], marker="s", color="C2")

```

This produces the obviously wrong result: 

![image](https://user-images.githubusercontent.com/18312003/111542111-3a0a4700-8769-11eb-9afb-7f2c002f213d.png)

I believe that it is quite easy to fix this by adding a boolean `kwarg` to both `overplot_lines` and `overplot_points` with `reverse=False` by default, and altering the loop if it is `True`. I could easily submit a PR to fix this if it would be helpful. However, it may be possible to get information from the plot itself in `overplot_*` and see if it was created with `reverse=True`, but that is a bit more involved and I am not completely sure how general it would be -- thoughts welcome!

**Comment by NeutralKaon** [comment]
And, just in case this is useful to anyone else, the easiest work around is by just iterating backwards: 

```python3
figure = corner.corner(samples,reverse=True)
axes = np.array(figure.axes).reshape((ndim, ndim))

# Loop over the diagonal
for i in range(ndim):
    ax = axes[ndim-i-1, ndim-i-1]
    ax.axvline(value1[i], color="g")
    ax.axvline(value2[i], color="r")
    
# Loop over the histograms
for yi in range(ndim):
    for xi in range(yi+1):
        ax = axes[ndim - yi - 1, ndim - xi -1] #axes[yi, xi]
        print(xi,yi,ex,ey)
        ax.axvline(value1[xi], color="g")
        ax.axvline(value2[xi], color="r")
        ax.axhline(value1[yi], color="g")
        ax.axhline(value2[yi], color="r")
        ax.plot(value1[xi], value1[yi], "sg")
        ax.plot(value2[xi], value2[yi], "sr")
```
This produces the expected result: 
![image](https://user-images.githubusercontent.com/18312003/111546523-b7d15100-876f-11eb-914f-39c5604748a1.png)

I think that patching as per the following might be a vaguely sensible way of doing this: 

```python3 
diff --git a/src/corner/core.py b/src/corner/core.py
index 8948839..7e5674f 100644
--- a/src/corner/core.py
+++ b/src/corner/core.py
@@ -683,7 +683,7 @@ def hist2d(
     _set_ylim(new_fig, ax, range[1])


-def overplot_lines(fig, xs, **kwargs):
+def overplot_lines(fig, xs, reverse=False, **kwargs):
     """
     Overplot lines on a figure generated by ``corner.corner``

@@ -698,6 +698,10 @@ def overplot_lines(fig, xs, **kwargs):
        call that originally generated the figure. The entries can optionally
        be ``None`` to omit the line in that axis.

+    reverse: bool
+       A boolean flag that should be set to 'True' if the corner plot itself
+       was plotted with 'reverse=True'.
+
     **kwargs
         Any remaining keyword arguments are passed to the ``ax.axvline``
         method.
@@ -705,17 +709,28 @@ def overplot_lines(fig, xs, **kwargs):
     """
     K = len(xs)
     axes = _get_fig_axes(fig, K)
-    for k1 in range(K):
-        if xs[k1] is not None:
-            axes[k1, k1].axvline(xs[k1], **kwargs)
-        for k2 in range(k1 + 1, K):
+    if reverse:
+        for k1 in range(K):
+            if xs[k1] is not None:
+                axes[K-k1-1, K-k1-1].axvline(xs[k1], **kwargs)
+            for k2 in range(k1 + 1, K):
+                if xs[k1] is not None:
+                    axes[K-k2-1, K-k1-1].axvline(xs[k1], **kwargs)
+                if xs[k2] is not None:
+                    axes[K-k2-1, K-k1-1].axhline(xs[k2], **kwargs)
+
+    else:
+        for k1 in range(K):
             if xs[k1] is not None:
-                axes[k2, k1].axvline(xs[k1], **kwargs)
-            if xs[k2] is not None:
-                axes[k2, k1].axhline(xs[k2], **kwargs)
+                axes[k1, k1].axvline(xs[k1], **kwargs)
+            for k2 in range(k1 + 1, K):
+                if xs[k1] is not None:
+                    axes[k2, k1].axvline(xs[k1], **kwargs)
+                if xs[k2] is not None:
+                    axes[k2, k1].axhline(xs[k2], **kwargs)


-def overplot_points(fig, xs, **kwargs):
+def overplot_points(fig, xs, reverse=False, **kwargs):
     """
     Overplot points on a figure generated by ``corner.corner``

@@ -729,6 +744,10 @@ def overplot_points(fig, xs, **kwargs):
        that is compatible with the :func:`corner.corner` call that originally
        generated the figure.

+    reverse: bool
+       A boolean flag that should be set to 'True' if the corner plot itself
+       was plotted with 'reverse=True'.
+
     **kwargs
         Any remaining keyword arguments are passed to the ``ax.plot``
         method.
@@ -739,9 +758,15 @@ def overplot_points(fig, xs, **kwargs):
     xs = _parse_input(xs)
     K = len(xs)
     axes = _get_fig_axes(fig, K)
-    for k1 in range(K):
-        for k2 in range(k1 + 1, K):
-            axes[k2, k1].plot(xs[k1], xs[k2], **kwargs)
+    if reverse:
+        for k1 in range(K):
+            for k2 in range(k1):
+                 axes[K - k1 - 1, K - k2 - 1].plot(xs[k1], xs[k2], **kwargs)
+
+    else:
+         for k1 in range(K):
+             for k2 in range(k1 + 1, K):
+                 axes[k2, k1].plot(xs[k1], xs[k2], **kwargs)


 def _parse_input(xs):
```

**Comment by NeutralKaon** [comment]
Fixed in #156 -- thanks Dan! :-)
